src/mmw/apps/geoprocessing_api/views.py

```diff
@@ -85,7 +90,9 @@ def start_analyze_soil(request, format=None):
 @decorators.permission_classes((AllowAny, ))
 def start_analyze_animals(request, format=None):
     user = request.user if request.user.is_authenticated() else None
-    area_of_interest, __ = parse_input(request.data['analyze_input'])
+
+    wkaoi_str = request.query_params.get('wkaoi', None)
```

Comment:
Seems strange to parse out a `wkaoi_str` if we're going to ignore the results of the lookup done in `parse_input`.  Can this be removed in all the non-raster jobs since `wkaoi_str` has a default value of `None` in the function?

Reply:
In the case where no area of interest was sent, we use the `wkaoi_str` to look up the aoi, whether the job is raster or not.

Even though we don't benefit from the wkaoi caching for these jobs, I think we should still accept the wkaoi id as input. This makes our requests on the app more uniform and makes it so the user doesn't have to keep track of which jobs are raster and which aren't.

Reply:
I see.  I'm not advocating for changing the request shape, but the method reads like it does not utilize the argument based on it ignoring the wkaoi return.  I think an improvement might be to rename things from `parse_*` to `load_*` or similar, since it's actually fetching data, not just parsing input.  Also, the second return variable which is only occasionally used appears to either be `None` or the id passed in as an argument initially.  Seems like this can be removed (since the caller necessarily has it), so it's clear that this function _only_ returns an area of interest, regardless of what mixture of inputs is received.  It reads like it returns something special only if wkaoi is provided.